File: PyProject/leetcode/remove-duplicates-from-sorted-array.py
from typing import List


class RemoveDuplicatesFromSortedArray:
    def removeDuplicates(self, nums: List[int]) -> int:
        start = 0
        end = 1
        length = len(nums)
        while end < length - 1:
            if (end + 1) >= length:
                break
            if nums[end] == nums[end + 1]:
                end += 1
                continue
            else:
                nums[start + 1] = nums[end + 1]
                start += 1
                end += 1
        return start

File: GoProject/leetcode/array/remove-duplicates-from-sorted-array.go
package array

func removeDuplicates(nums []int) int {
	slow, quick := 0, 0

	l := len(nums)
	for i := 0; i < l-1; i++ {
		quick += 1
		if nums[slow] == nums[quick] {
			continue
		} else {
			slow += 1
			if slow == quick {
				continue
			} else {
				nums[slow] = nums[quick]
			}
		}
	}

	return slow+1
}
